exonum-java-binding/core/rust/integration_tests/tests/runtime.rs

```diff
@@ -0,0 +1,667 @@
+        .with_instances(instances);
+    assert_panics(
+        format!(
+            "{} {}",
+            "Could not find a suitable constructor in",
+            "com.exonum.binding.fakes.services.invalidservice.NonInstantiableService",
+        )
+        .as_str(),
+        || test_kit_builder.create(),
+    );
+}
+
+#[test]
+fn service_can_modify_db_on_initialize() {
+    let (runtime, _instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+    test_kit.create_block();
+
+    let snapshot = test_kit.snapshot();
+    let test_map = create_init_service_test_map(&*snapshot, fake_service::TEST_SERVICE_NAME);
+    let key = hash(fake_service::INITIAL_ENTRY_KEY.as_ref());
+    let value = test_map
+        .get(&key)
+        .expect("Failed to find the entry created in the test service");
+    assert_eq!(fake_service::INITIAL_ENTRY_VALUE, value);
+
+    test_kit.stop();
+}
+
+#[test]
+fn execute_valid_transaction() {
+    let tx_args = &[1_u8, 2_u8, 3_u8, 4_u8];
+    let tx_args_hex = hex::encode(tx_args);
+
+    let (runtime, instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+
+    test_kit.create_block();
+
+    let tx = create_transaction(instance_id, fake_service::SET_ENTRY_TX, tx_args);
+    let block = test_kit.create_block_with_transaction(tx);
+
+    assert_eq!(block.transactions.len(), 1);
+    assert!(block.transactions.get(0).unwrap().status().is_ok());
+
+    test_kit.create_block();
+
+    let snapshot = test_kit.snapshot();
+    let test_entry = create_tx_test_entry(&*snapshot, fake_service::TEST_SERVICE_NAME);
+    let value = test_entry
+        .get()
+        .expect("Failed to find the entry created in the test service");
+
+    assert_eq!(tx_args_hex, value);
+
+    test_kit.stop();
+}
+
+#[test]
+fn submit_non_convertible_tx() {
+    let tx_args = &[1_u8, 2_u8, 3_u8, 4_u8];
+
+    let (runtime, instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+    test_kit.create_block();
+
+    let tx = create_transaction(instance_id, fake_service::THROW_SOE_TX, tx_args);
+    let block = test_kit.create_block_with_transaction(tx);
+
+    assert_eq!(block.transactions.len(), 1);
+    let status = block.transactions.get(0).unwrap().status();
+    assert!(status.is_err());
+    assert!(status
+        .unwrap_err()
+        .to_string()
+        .contains("java.lang.StackOverflowError;"));
+
+    test_kit.stop();
+}
+
+#[test]
+fn submit_service_error_on_exec_tx() {
+    let tx_args = &[255_u8, 2_u8, 3_u8, 4_u8];
```

Comment:
Why is there anything in the arguments atop of the error code?

Reply:
Also, it would fail if tests were active (Java error codes shall not exceed 127)